cypress/integration/specialFeatures/scriptSwitching/testsForCanonicalOnly.js

```diff
@@ -0,0 +1,75 @@
+import visitPage from '../../../support/helpers/visitPage';
+import {
+  allVariantAssertions,
+  nonCookieVariantAssertions,
+} from '../utilities/scriptSwitchingJourneyAssertions';
+import {
+  clickScriptSwitcher,
+  clickHomePageLink,
+  clickPromoLinkOnHomePage,
+} from '../utilities/scriptSwitchingJourneyActions';
+import {
+  getPrivacyBannerAccept,
+  getCookieBannerAccept,
+} from '../utilities/cookiePrivacyBanner';
+
+export default ({
+  serviceId,
+  serviceName,
+  pageType,
+  path,
+  variant,
+  otherVariant,
+}) => {
+  describe(`Script Switching - ${serviceId} - ${pageType} - ${path}`, () => {
+    beforeEach(() => {
+      cy.clearCookies();
+      visitPage(path, pageType);
+    });
+
+    it(`should change to the correct script when switching script between ${variant} and ${otherVariant}`, () => {
+      // Accept privacy banner
+      getPrivacyBannerAccept(serviceId, variant).click();
+
+      // Accept cookie banner
+      getCookieBannerAccept(serviceId, variant).click();
+
+      cy.log(
+        `Asserting script switch button, url and document lang against variant: ${variant}`,
+      );
+      nonCookieVariantAssertions(serviceName, variant);
+
+      // Clicks script switcher
+      clickScriptSwitcher(otherVariant);
```

Comment:
I am wondering if we can / should delete the param from this, and just assume that we can click that button, regardless of what the data variant is, and that the assertions will catch any problems?

Reply:
Good point, it is a bit belt-and-braces, and **is** covered by the script button assertion

Reply:
Akh the service name is used to get the correct button. Nothing else in there that is specific to that element.

![Screenshot 2020-04-29 at 12 47 53](https://user-images.githubusercontent.com/9042672/80592666-b6901c00-8a17-11ea-9f23-68b3189f6b12.png)